### PR TITLE
feat: alpha release

### PR DESCRIPTION
**Pull request by RobertCraigie**
Among many other smaller cleanups, this PR migrates from `node-fetch` to builtin fetch.

These changes will be released under the `0.34.0-alpha.0` version.

Report issues / feedback here: https://github.com/anthropics/anthropic-sdk-typescript/issues/645

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -1,6 +1,6 @@
 ## Setting up the environment
 
-This repository uses [`yarn@v1`](https://classic.yarnpkg.com/lang/en/docs/install/#mac-stable).
+This repository uses [`yarn@v1`](https://classic.yarnpkg.com/lang/en/docs/install).
```
Comment (michael-cohen-io) [inline CONTRIBUTING.md:3]: why the change here?

File: src/api-promise.ts [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+   * returned via the `request-id` header which is useful for debugging requests and resporting
+   * issues to Anthropic.
+   *
+   * If you just want to get the raw `Response` instance without parsing it,
+   * you can use {@link asResponse()}.
+   *
+   * 👋 Getting the wrong TypeScript type for `Response`?
+   * Try setting `"moduleResolution": "NodeNext"` or add `"lib": ["DOM"]`
+   * to your `tsconfig.json`.
+   */
```
Comment (michael-cohen-io) [inline src/api-promise.ts:64]: 👏

**Review by stephaniepang97 — APPROVED** [review]

**Review by michael-cohen-io — APPROVED** [review]

File: CONTRIBUTING.md (hunk 1 above)
Comment (RobertCraigie) [reply CONTRIBUTING.md:3]: old link was outdated :)
